### DISCUSSION TITLE
Renovate and GitHub Packages

### DISCUSSION THREAD

**Discussion by Jamesking56** [renovatebot/renovate#12289] [Request Help]
Hi,

Using Renovate via the official GitHub App.

My repo I'm configuring Renovate on has a `Dockerfile` that uses a private GitHub Packages image in the `FROM`. Renovate is warning me about this in the configuration PR:

> ### ⚠ Dependency Lookup Warnings ⚠
> 
> Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.
> 
> -   `Failed to look up dependency ghcr.io/momentumstudioltd/laravel`
> 
> Files affected: `.ci/Dockerfile`

GitHub actions uses this `Dockerfile` as well, so is it possible for Renovate to manage updates for this or to give the GitHub App permission to GitHub Packages?

**Comment by rarkins** [comment]
First part: Can GitHub apps access packages at all?

It didn't used to be the case: https://github.community/t/using-github-packages-with-github-apps/130950/2

Same answer from February 2021: https://github.community/t/access-github-packages-using-github-app-token/161793/2

Not sure if this proves anything: https://docs.github.com/en/rest/overview/endpoints-available-for-github-apps

Yet there is an option for App owners to request permissions for Packages (we don't currently):

![image](https://user-images.githubusercontent.com/6311784/138654497-08b6e8bd-6a62-4bdb-b6d4-b61cd814634f.png)

I don't want to request that until we are confident that it would *work* out of the box. We may need some code changes, e.g. in hostRules to add ghcr.io 

Otherwise it seems like you'd need to use a PAT in a hostRules entry specifically for the packages host.

**Reply by jamesdh** [reply]
Just wanted to add to this that we use our own private GitHub App to run renovate in GitHub Actions. I tried granting read-only access to Packages via the app permissions, and it's still unable check for package updates in the GitHub Package Repository, continuing to get...

> Dependency lookup unauthorized. Please add authentication with a hostRule
       "failedUrl": "[https://maven.pkg.github.com]

**Reply by Jamesking56** [reply]
This is actually fixed and working in the official GitHub app, I don't remember how I fixed it - I'm starting to think Renovate actually fixed it?

**Reply by rarkins** [reply]
@Jamesking56 thanks for the feedback. Could you confirm:
- have you made any particular host rules or package rules settings to get it to work?
- is it any particular type of package it works for you, Eg npm,  composer, maven?

**Reply by Jamesking56** [reply]
I think what I must've done is added a GitHub token with access to packages as follows:

```
"extends": {
    ":githubComToken(<TOKEN>)"
}
```

But Renovate does create PRs when there is a newer tag of my custom images in Packages now

I am specifically using Docker images in Packages here which are private to my org

**Reply by anthonator** [reply]
@Jamesking56 can you provide a more details explanation on what worked for you? We're trying to access a private image hosted on ghcr.io without luck. I've tried using `:githubComToken` but still getting the same `Failed to look up docker dependency` error.

**Reply by Jamesking56** [reply]
@anthonator I don't think the fix works anymore as I've got the same error come back, see the newer discussion: https://github.com/renovatebot/renovate/discussions/17363

**Reply by anthonator** [reply]
Dang! Thank you for your response @Jamesking56!
